Clojure/tests/syntax_test_clojure.clj

```diff
@@ -0,0 +1,2007 @@
+; SYNTAX TEST "Packages/Clojure/Clojure.sublime-syntax"
+
+; # Comments and whitespace
+
+  ;blah
+; ^ comment.line.clojure punctuation.definition.comment
+;  ^^^^ comment.line.clojure
+
+  ;;; blah
+; ^^^ comment.line.clojure punctuation.definition.comment
+;    ^^^^^ comment.line.clojure
+
+  blah;blah;blah
+; ^^^^- comment
+;     ^ comment.line.clojure
+
+  #!blah
+; ^^ comment.line.clojure punctuation.definition.comment
+;   ^^^^^ comment.line.clojure
+  #! blah
+; ^^ comment.line.clojure punctuation.definition.comment
+;   ^^^^^^ comment.line.clojure
+  #!#!#! blah
+; ^^ comment.line.clojure punctuation.definition.comment
+;   ^^^^^^^^^^ comment.line.clojure
+
+  blah,blah, blah
+;     ^ comment.punctuation.comma.clojure
+;      ^- comment
+;          ^ comment.punctuation.comma.clojure
+;           ^- comment
+
+; ## Include end-of-line
+
+; ; blah
+;^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.clojure
+
+
+
+; # Constants
+
+  true false nil
+; ^^^^ constant.language.clojure
+;     ^- constant
+;      ^^^^^ constant.language.clojure
+;           ^- constant
+;            ^^^ constant.language.clojure
+
+; ## Breaks
+
+  true,false,nil
+; ^^^^ constant.language.clojure
+;     ^ comment.punctuation.comma.clojure
+  true;false;nil
+; ^^^^ constant.language.clojure
+;     ^ comment.line.clojure punctuation.definition.comment
+
+; ## Unaffected
+
+  'nil (true) (nil)
+; ^ constant.other.symbol.reader-macro.clojure
+;  ^^^ constant.language.clojure
+;      ^ punctuation.section.parens.begin.clojure
+;       ^^^^ constant.language.clojure
+;           ^ punctuation.section.parens.end.clojure
+
+; ## No highlighting
+
+  nill nil- -nil nil?
+; ^^^^^^^^^^^^^^^^^^^- constant
+
+
+
+; # Numbers
+
+  1234 1234N +1234 +1234N -1234 -1234N
+; ^^^^ constant.numeric.clojure
+;     ^- constant
+;      ^^^^^ constant.numeric.clojure
+;            ^^^^^ constant.numeric.clojure
+;                  ^^^^^^ constant.numeric.clojure
+;                         ^^^^^ constant.numeric.clojure
+;                               ^^^^^^ constant.numeric.clojure
+  0x1234af 0x1234afN 0X1234AF 0X1234AFN
```

Comment:
What do the various `af` and `n` suffixes mean? We have some conventions we've been applying to recent work on syntaxes where we apply scopes to target storage/type info in the numeric literals. My hunch is they may be applicable here also. See https://github.com/sublimehq/Packages/blob/master/Python/syntax_test_python.py#L935-L940 for an example.

Reply:
`0x_______` is a hex integer literal, which may include `[A-Fa-f]`.

Reply:
Ooh I see about storage. By that logic, the `N` at the end should be `storage.type.numeric.bigint.clojure`. I think it should still be part of the `constant.numeric` scope, since it's part of the number literal, so it needs a double scope.

Reply:
Yes, that's also how it is in Python (and other syntaxes). Also note the `punctuation.definition.numeric.hexadecimal.python` scopes.